state-chain/runtime/src/chainflip/simulate_swap.rs

```diff
@@ -0,0 +1,265 @@
+			},
+			ForeignChainAndAsset::Assethub(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, AssethubInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees, fees_withheld)
+			},
+		}
+	}
+
+	let include_fee = |fee_type: FeeTypes| !exclude_fees.contains(&fee_type);
+
+	// Default to using the DepositChannel fee unless specified.
+	let (mut amount_to_swap, ingress_fee) = if include_fee(FeeTypes::IngressDepositChannel) {
+		remove_fees(IngressOrEgress::IngressDepositChannel, input_asset, input_amount)
+	} else if include_fee(FeeTypes::IngressVaultSwap) {
+		remove_fees(IngressOrEgress::IngressVaultSwap, input_asset, input_amount)
+	} else {
+		(input_amount, 0u128)
+	};
+
+	// If no DCA parameter is given, swap the entire amount with 1 chunk.
+	let number_of_chunks: u128 = dca_parameters
+		.map(|dca| sp_std::cmp::max(dca.number_of_chunks, 1u32))
+		.unwrap_or(1u32)
+		.into();
+
+	let mut fees_vec = vec![];
+
+	// Calculate the network fee in both USDC and in terms of the input asset.
+	// We manually calculate the minimum network fee to avoid complications with the network fee
+	// minimum effecting the simulated chunk.
+	let (network_fee_amount_input_asset, network_fee_usdc) = if include_fee(FeeTypes::Network) {
+		let network_fee = pallet_cf_swapping::Pallet::<Runtime>::get_network_fee_for_swap(
+			input_asset,
+			output_asset,
+			is_internal,
+		);
+
+		let min_network_fee_input_asset =
+			pallet_cf_swapping::Pallet::<Runtime>::calculate_input_for_desired_output(
+				output_asset,
+				Asset::Usdc,
+				network_fee.minimum,
+				false, // Do not apply network fee to this calculation
+			)
+			.unwrap_or_default();
+		let network_fee_input_asset = network_fee.rate * amount_to_swap;
+		if min_network_fee_input_asset > network_fee_input_asset {
+			(min_network_fee_input_asset, network_fee.minimum)
+		} else {
+			let amount_per_chunk: u128 = amount_to_swap / number_of_chunks;
+			let swap_output = &Swapping::simulate_swaps(vec![Swap::new(
+				Default::default(), // Swap id
+				Default::default(), // Swap request id
+				input_asset,
+				output_asset,
+				amount_per_chunk,
+				None, // Refund params
+				vec![FeeType::NetworkFee(NetworkFeeTracker::new_without_minimum(
+					network_fee.clone(),
+				))],
+				Default::default(), // Execution block
+			)])
+			.map_err(|_| DispatchError::Other("Failed to calculate network fee"))?;
+
+			(
+				network_fee_input_asset,
+				swap_output[0].network_fee_taken.unwrap_or_default() * number_of_chunks,
+			)
+		}
+	} else {
+		(0, 0)
+	};
+
+	amount_to_swap.saturating_reduce(network_fee_amount_input_asset);
+
+	let amount_per_chunk: u128 = amount_to_swap / number_of_chunks;
+
+	if broker_commission > 0 {
+		let fee = FeeType::BrokerFee(
+			vec![Beneficiary { account: AccountId::new([0xbb; 32]), bps: broker_commission }]
```

Comment:
I think you can construct `bounded_vec!` directly here without having to convert?